dlk/python/dlk/templates/include/func/resize_nearest_neighbor.h

```diff
@@ -0,0 +1,118 @@
+/* Copyright 2018 The Blueoil Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef DLK_FUNC_RESIZE_NEAREST_NEIGHBOR_H_INCLUDED
+#define DLK_FUNC_RESIZE_NEAREST_NEIGHBOR_H_INCLUDED
+
+#include "global.h"
+#include "time_measurement.h"
+#include "tensor_view.h"
+
+
+inline void func_ResizeNearestNeighbor(const TensorView<float, MemoryLayout::NHWC>& input,
+    const TensorView<float, MemoryLayout::NHWC>& output) {
+
+  Measurement::Start("ResizeNearestNeighbor");
+
+  const auto in_shape = input.get_shape();
+  const auto in_height = in_shape[1];
+  const auto in_width = in_shape[2];
+
+  const auto out_shape = output.get_shape();
+  const auto out_height = out_shape[1];
+  const auto out_width = out_shape[2];
+  const auto out_depth = out_shape[3];
+
+  const auto height_scale = (in_height << 16) / out_height + 1;
```

Comment:
Sorry, I don't understand why this `<< 16` bitshifting is necessary. Could you explain a bit?

Reply:
Since shape has a type `size_t`, If `in_height / out_height < 0`, it will get a `0`. Here it uses a `fixed-point` style to address this issue.

Reply:
thanks, understood.